src/combinatorics/simple_graph/adj_matrix.lean

```diff
@@ -15,89 +16,256 @@ properties to computational properties of the matrix.
 
 ## Main definitions
 
-* `adj_matrix` is the adjacency matrix of a `simple_graph` with coefficients in a given semiring.
+* `matrix.adj_matrix`: `A : matrix V V α` is qualified as an "adjacency matrix" if
+  (1) every entry of `A` is `0` or `1`,
+  (2) `A` is symmetric,
+  (3) every diagonal entry of `A` is `0`.
+
+* `matrix.adj_matrix.to_graph`: for `A : matrix V V α` and `h : adj_matrix A`,
+  `h.to_graph` is the simple graph induced by `A`.
+
+* `matrix.compl`: for `A : matrix V V α`, `A.compl` is supposed to be
+  the adjacency matrix of the complement graph of the graph induced by `A.adj_matrix`.
+
+* `simple_graph.adj_matrix`: the adjacency matrix of a `simple_graph`.
 
 -/
 
 open_locale big_operators matrix
 open finset matrix simple_graph
 
 universes u v
-variables {α : Type u}
-variables (R : Type v) [semiring R]
+variables {V α β R : Type u}
+
+namespace matrix
+
+/-- `A : matrix V V α` is qualified as an "adjacency matrix" if
+    (1) every entry of `A` is `0` or `1`,
+    (2) `A` is symmetric,
+    (3) every diagonal entry of `A` is `0`. -/
+structure adj_matrix [has_zero α] [has_one α] (A : matrix V V α) : Prop :=
+(zero_or_one : ∀ i j, (A i j) = 0 ∨ (A i j) = 1 . obviously)
+(symm : A.is_symm . obviously)
+(loopless : ∀ i, A i i = 0 . obviously)
+
+namespace adj_matrix
+
+lemma apply_diag [has_zero α] [has_one α]
+{A : matrix V V α} (h : adj_matrix A) (i : V) :
+  A i i = 0 :=
+by simp [h.loopless]
+
+@[simp]
+lemma apply_diag_ne [mul_zero_one_class α] [nontrivial α]
+{A : matrix V V α} (h : adj_matrix A) (i : V) :
+  ¬ A i i = 1 :=
+by simp [h.apply_diag i]
+
+@[simp]
+lemma apply_ne_one_iff [mul_zero_one_class α] [nontrivial α]
+{A : matrix V V α} (h : adj_matrix A) (i j : V) :
+  ¬ A i j = 1 ↔ A i j = 0 :=
```

Comment:
```suggestion
  A i j ≠ 1 ↔ A i j = 0 :=
```
Unless the linter doesn't allow that

Reply:
I tested this. `#lint` does report an error if using `ne`.